doc/decisions/STEPS.md

```diff
@@ -0,0 +1,98 @@
+# Steps
+
+This document describes all steps a decision can run through.
+
+```mermaid
+flowchart LR
+    s((Start)) --> Drafts --> In_Discussion --> In_Progress --> Decided --> Partially_Implemented --> Implemented
+
+    %% Shortcuts:
+    s --> Decided
+    Decided --> Implemented
+```
+
+Additionally, decision that are not yet "Decided" can be become "Rejected" or "Delayed" at any point.
+
+> The first PR for a decision must always create the decision in the "Drafts" state.
+> If during the reviews of this PR it becomes clear that the decision is further along, can be moved to "In Discussion" or "In Progress" before the merge.
+
+## Drafts
+
+> This step is highly recommended and it is even required if the problem is not yet clear to all the core developers.
+
+The first step is to create a PR with:
+
+- **one** decision, where at least the "Problem" is filled out and "Decision", "Rationale" and "Implications" are **not** yet filled out.
+- a link from [README.md](../README.md) from the "Drafts" section to this decision.
+- optional backlinks from related decisions.
+
+This step is very important:
+
+- it is brainstorming for completely new ideas.
+- it clarifies the scope of the decision.
+- it clarifies relation to other problems.
+
+Decisions will have much smoother further steps if this step is done carefully without prejudice.
+It is especially important that one shouldn't have a fixed mind-set about a preferred solution from the beginning.
+
+> Everyone must agree that the problem exists so that a decision PR in "Drafts" step can be merged.
+> At least the problem must be clear to everyone involved before the decision can leave the "Drafts" step.
+> It must be so clear that everyone would be able to describe a test case that shows if a solution fixes the problem.
+
+## In Discussion
+
+> This step is recommended if it is not yet clear to the core developers which solution is the best.
+
+Here you must ensure:
+
+- problem, constraint and assumptions are well-explained and sound
+- consistency with other decisions
+- links from/to related decisions are created
+- there are several considered alternatives, each with rationale and implication
+- "Decision", "Rationale" and "Implications" are **not** yet filled out if there are people arguing for different options
+
+Here the decision should not only have one decision but should describe several solutions.
+For each solution a proposal, rationale and optionally implications should be given.
+
+## In Progress
+
+- You must include all further alternative proposals made in the "Considered Alternatives" section.
+- Now it is allowed to have the decision from the previous round in the "Decision" section.
+
+> For decisions "In Progress", unlike "In Discussion", also the solution space next to the problem is clear.
+> It is the highest of the steps reachable for a decision PR that started in "Drafts".
```

Comment:
IMO that sentence is pretty confusing.

For one, I'd say "Decided" is the highest reachable step. Because I could create a PR with the decision as "Drafts" and move it trough the stages. Only when it reaches "Decided" do we require a merge, before we can move on.

Also the bit about "started in Drafts" is confusing. If I start the decision in "In Discussion" I could move beyond this step, or what?

Reply:
There is also the rule: "If during the reviews of this PR it becomes clear that the decision is further along, can be moved to "In Discussion" or "In Progress" before the merge." (which is the full list), so it is not possible to directly go to Decided from Drafts. See also updated Diagram: 674ec185e894c13cc22edb5d547352ef6ef70e8a

I changed the confusing sentence to: "Without merges in between, this is the last step reachable for a decision PR that started in "Drafts"." 570e48ab307995a791d7340d62cbbd23139e1331